src/actionspace/ObjectOrientedActionSpace.cpp

```diff
@@ -1,178 +1,157 @@
-    }
-}
-
-void ObjectOrientedActionSpace::find_start_pos(
-    const Eigen::Vector3d& obj_pose,
-    const double& edge_offset,
-    const double& heading,
-    Eigen::Vector2d* start_pos) const
-{
-    start_pos->x() = obj_pose.x() - center_offset * cos(heading) +
-                     edge_offset * sin(heading);
-    start_pos->y() = obj_pose.y() - center_offset * sin(heading) -
-                     edge_offset * cos(heading);
-}
-
-void ObjectOrientedActionSpace::generate_actions(
-    const Eigen::Vector3d& obj_pose,
-    const double& edge_offset)
-{
-    std::vector<double> cube_offsets = num_offset == 1 ?
-        std::vector<double>{0} :
-        utils::linspace(-edge_offset, edge_offset, num_offset);
-
-    std::vector<double> headings;
-    find_headings(obj_pose(2), &headings);
-
-    int action_id = 0;
-    for (const auto& heading : headings) {
-        for (const auto& cube_offset : cube_offsets) {
-            for (const auto& speed : speeds) {
-                for (const auto& duration : durations) {
-                    Eigen::Vector2d start_pos;
-                    find_start_pos(obj_pose, cube_offset, heading, &start_pos);
-
-                    ObjectOrientedActionSpace::Action* action =
-                        static_cast<ObjectOrientedActionSpace::Action*>(
-                            get_action(action_id));
-                    if (action == nullptr) {
-                        actions.push_back(new Action(
-                            speed,
-                            duration,
-                            Eigen::Vector3d(
-                                start_pos(0),
-                                start_pos(1),
-                                heading),
-                            -cube_offset / edge_offset
-                            ));
-                    } else {
-                        action->update_action(
-                            speed,
-                            duration,
-                            Eigen::Vector3d(
-                                start_pos(0),
-                                start_pos(1),
-                                heading),
-                            -cube_offset / edge_offset
-                            );
-                    }
-                    action_id++;
-                }
-            }
-        }
-    }
-}
-
 ActionSpace::Action* ObjectOrientedActionSpace::get_action(
-    const int& action_id) const
-{
+    const int& action_id) const {
     if (!is_valid_action_id(action_id)) {
         return nullptr;
     }
-    return actions[action_id];
+    return m_actions[action_id];
 }
 
 bool ObjectOrientedActionSpace::is_valid_action_id(
-    const int& action_id) const
-{
-    return action_id < actions.size() && action_id >= 0;
+    const int& action_id) const {
+    return action_id < m_actions.size() && action_id >= 0;
+}
+
+void ObjectOrientedActionSpace::generic_to_object(
+    const GenericAction& _action,
```

Comment:
should take an action_id

Reply:
The function should call the `get_generic_action` function with the given ID to get the generic action